### PR TITLE
Command line tool to reset replica tokens in Azure

### PR DESCRIPTION
**Pull request by lightningrob**

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java [hunk 1]
```diff
@@ -128,25 +133,25 @@
...
-  @Config("clustermap.host.name")
+  @Config(CLUSTERMAP_HOST_NAME)
   public final String clusterMapHostName;
 
   /**
    * The port number associated with this node.
    */
-  @Config("clustermap.port")
+  @Config(CLUSTERMAP_PORT)
   @Default("null")
```
Comment (ankagrawal) [inline ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:155]: Maybe we can either remove Default from here, or change it to empty string.

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java [hunk 2]
```diff
@@ -129,6 +129,10 @@
     batchTimeout = null;
   }
 
+  public BlobServiceClient getStorageClient() {
```
Comment (ankagrawal) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java:132]: javadocs.

File: ambry-tools/src/main/java/com/github/ambry/tools/admin/AzureTokenResetTool.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      int tokensDeleted = resetTokens(clusterName);
+      logger.info("Deleted tokens for {} partitions", tokensDeleted);
+    } catch (Exception ex) {
+      ex.printStackTrace();
+      logger.error("Command {} failed", commandName, ex);
+      System.exit(1);
+    }
+  }
+
+  // Reset the offset token by deleting the blob from the container
```
Comment (ankagrawal) [inline ambry-tools/src/main/java/com/github/ambry/tools/admin/AzureTokenResetTool.java:73]: Nit: Not sure about how we maintain javadocs for tools. But if we maintain it like prod code, then we might need javadocs for methods in this class.

File: ambry-tools/src/main/java/com/github/ambry/tools/admin/AzureTokenResetTool.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  public static int resetTokens(String containerPrefix) throws BlobStorageException {
+    AtomicInteger tokensDeleted = new AtomicInteger(0);
```
Comment (ankagrawal) [inline ambry-tools/src/main/java/com/github/ambry/tools/admin/AzureTokenResetTool.java:75]: Curious as to why we need AtomicInteger here? Does forEachRemaining for PagedIterable's returned by listBlobContainers, return a parallel stream?

File: ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java [hunk 5]
```diff
@@ -282,10 +287,10 @@ public ClusterMapConfig(VerifiableProperties verifiableProperties) {
     clusterMapClusterChangeHandlerType =
         verifiableProperties.getString("clustermap.cluster.change.handler.type", "SimpleClusterChangeHandler");
     clusterMapDcsZkConnectStrings = verifiableProperties.getString("clustermap.dcs.zk.connect.strings", "");
```
Comment (jsjtzyy) [inline ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:289]: minor: in future PR, we can replace all config names with string constants (like what you did in this PR to make them consistent)

Comment (lightningrob) [reply ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:289]: Yes.  I didn't do it in this PR to keep the scope limited.

File: ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java (hunk 1 above)
Comment (lightningrob) [reply ambry-api/src/main/java/com/github/ambry/config/ClusterMapConfig.java:155]: I don't know why it's there, but I'd prefer not to touch it right now.

File: ambry-tools/src/main/java/com/github/ambry/tools/admin/AzureTokenResetTool.java (hunk 4 above)
Comment (lightningrob) [reply ambry-tools/src/main/java/com/github/ambry/tools/admin/AzureTokenResetTool.java:75]: I had it as an int but the compiler complained that I couldn't pass a non-final variable to a lambda, and this seemed like the easiest way to work around it.

**Pull request merged**